source/extensions/filters/network/dubbo_proxy/metadata.h

```diff
@@ -19,104 +19,63 @@ namespace DubboProxy {
-    parameter_map_->emplace(index, value);
-  }
-  const std::string& getParameterValue(uint32_t index) const {
-    if (parameter_map_) {
-      auto itor = parameter_map_->find(index);
-      if (itor != parameter_map_->end()) {
-        return itor->second;
-      }
-    }
-
-    return EMPTY_STRING;
+  template <typename T = SerializationType> void setSerializationType(T type) {
+    ASSERT((std::is_same<uint8_t, typename std::underlying_type<T>::type>::value));
+    serialization_type_ = static_cast<uint8_t>(type);
   }
-  bool hasParameters() const { return parameter_map_ != nullptr; }
-  const ParameterValueMap& parameters() {
-    ASSERT(hasParameters());
-    return *parameter_map_;
+  template <typename T = SerializationType> T serialization_type() const {
+    ASSERT((std::is_same<uint8_t, typename std::underlying_type<T>::type>::value));
+    return static_cast<T>(serialization_type_);
   }
 
-  bool hasHeaders() const { return headers_ != nullptr; }
-  const Http::HeaderMap& headers() const {
-    ASSERT(hasHeaders());
-    return *headers_;
+  template <typename T = ResponseStatus> void setResponseStatus(T status) {
+    ASSERT((std::is_same<uint8_t, typename std::underlying_type<T>::type>::value));
+    response_status_ = static_cast<uint8_t>(status);
   }
-  void addHeader(const std::string& key, const std::string& value) {
-    assignHeaderIfNeed();
-    headers_->addCopy(Http::LowerCaseString(key), value);
-  }
-  void addHeaderReference(const Http::LowerCaseString& key, const std::string& value) {
-    assignHeaderIfNeed();
-    headers_->addReference(key, value);
+  template <typename T = ResponseStatus> T response_status() const {
+    ASSERT((std::is_same<uint8_t, typename std::underlying_type<T>::type>::value));
+    return static_cast<T>(response_status_.value());
   }
+  bool hasResponseStatus() const { return response_status_.has_value(); }
 
 private:
-  inline void assignHeaderIfNeed() {
-    if (!headers_) {
-      headers_ = std::make_unique<Http::HeaderMapImpl>();
-    }
-  }
-  inline void assignParameterIfNeed() {
-    if (!parameter_map_) {
-      parameter_map_ = std::make_unique<ParameterValueMap>();
-    }
-  }
-
   bool is_two_way_{false};
-  bool is_event_{false};
 
   MessageType message_type_{MessageType::Request};
-  SerializationType serialization_type_{SerializationType::Hessian};
-  absl::optional<ResponseStatus> response_status_;
+  ProtocolType proto_type_{ProtocolType::Dubbo};
 
-  int64_t request_id_ = 0;
+  absl::optional<uint8_t> response_status_;
+  absl::optional<uint32_t> timeout_;
+
+  RpcInvocationSharedPtr invocation_info_;
 
-  // Routing metadata.
-  std::string service_name_;
-  absl::optional<std::string> method_name_;
-  absl::optional<std::string> service_version_;
-  absl::optional<std::string> group_;
-  ParameterValueMapPtr parameter_map_;
-  HeaderMapPtr headers_; // attachment
+  uint8_t serialization_type_{static_cast<uint8_t>(SerializationType::Hessian2)};
```

Comment:
why not use `SerializationType`

Reply:
To be compatible with the internal HSF protocol, both HSF and Dubbo define SerializationType, but with different values.